core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -528,7 +522,7 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
     override def requestAndWaitForCommit() {
       this.synchronized {
         // only wait() if mirrorMakerConsumer has been initialized and it has not been cleaned up.
-        if (iter != null) {
+        while (iter != null) {
```

Comment:
Thanks, that's a good catch.  I added a counter to make sure that we don't get stuck waiting for someone else's invocation of `requestAndWaitForCommit` (which would reset the `immediateCommitRequest` variable back to true)